flink-runtime-web/web-dashboard/src/app/app.component.html

```diff
@@ -58,29 +58,31 @@ <h1>Apache Flink Dashboard</h1>
           </li>
         </ul>
       </li>
-      <li nz-menu-item routerLinkActive="ant-menu-item-selected" [routerLink]="['/task-manager']">
-        <span>
-          <i nz-icon nzType="schedule"></i>
-          <span>Task Managers</span>
-        </span>
-      </li>
-      <li nz-menu-item routerLinkActive="ant-menu-item-selected" [routerLink]="['/job-manager']">
-        <span>
-          <i nz-icon nzType="build"></i>
-          <span>Job Manager</span>
-        </span>
-      </li>
-      <li
-        *ngIf="webSubmitEnabled"
-        nz-menu-item
-        routerLinkActive="ant-menu-item-selected"
-        [routerLink]="['/submit']"
-      >
-        <span>
-          <i nz-icon nzType="upload"></i>
-          <span>Submit New Job</span>
-        </span>
-      </li>
+      <ng-container *ngIf="!historyServerEnv">
+        <li nz-menu-item routerLinkActive="ant-menu-item-selected" [routerLink]="['/task-manager']">
+          <span>
```

Comment:
nit: It's better to use CanActivate cancel the current navigation if users accesses these links under historyServerEnv, but even if not, there is no problem :)

Reply:
The purpose of hiding these menu items is to make our pages cleaner because JM and TM are unavailable under History Server (just an improvement). It is unnecessary to create a route guard to prevent entering these pages which should display nothing :)